### PR TITLE
Bring back auto-linking to Kernel and Kernel.SpecialForms

### PR DESCRIPTION
**Pull request by wojtekmach**

### CODE REVIEW

File: lib/ex_doc/autolink.ex [hunk 1]
```diff
@@ -393,10 +395,21 @@ defmodule ExDoc.Autolink do
 
     if Refs.public?(ref) do
       fragment(tool(module), kind, name, arity)
+    else
+      if kind == :function do
+        try_autoimported_function(name, arity, config)
+      end
```
Comment (josevalim) [inline lib/ex_doc/autolink.ex:401]:
```suggestion
    cond do
      Refs.public?(ref) -> fragment(tool(module), kind, name, arity)
      kind == :function -> try_autoimported_function(name, arity, config)
      true -> nil
    end
```

?

**Comment by eksperimental** [comment]
What was the reason behind for this feature to be initially removed?

**Comment by wojtekmach** [comment]
the thinking was to nudge people to use fully qualified name even for Kernel* functions as it it's more explicit and hopefully more approachable to beginners that may not be familiar with all built-in functions. But sometimes it is really useful to auto-link to these after all, I guess that's why we introduced this capability in the first place!

**Comment by eksperimental** [comment]
thank you @wojtekmach

**Pull request merged**